pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -535,70 +535,88 @@ func applyHTTPRouteDestination(
 		}
 	}
 
-	// TODO: eliminate this logic and use the total_weight option in envoy route
```

Comment:
Do we still want this TODO

Reply:
No. total_weights is deprecated. So we do not have to use.